### ISSUE TITLE
TreeModelFilter.refilter should take into consideration the fact that the view is empty

### ISSUE THREAD

**Issue by Steinhagen** [lutris/lutris#2767]
**Describe the bug**
When you are selecting a runner that doesn't have any visible entry, you'll get a number of GTK errors on your terminal output.

```
(lutris:320775): Gtk-CRITICAL **: 01:11:08.108: gtk_tree_model_sort_convert_iter_to_child_iter: assertion 'valid == TRUE' failed                                                                                                                      
(lutris:320775): Gtk-CRITICAL **: 01:11:08.108: gtk_tree_model_filter_real_unref_node: assertion 'filter->priv->stamp == iter->stamp' failed                                                                                                          
(lutris:320775): Gtk-CRITICAL **: 01:11:08.108: gtk_tree_model_sort_real_unref_node: assertion 'elt->ref_count > 0' failed
```
The error messages are caused by calling refilter on a runner with no elements if the previous runner had at least 1 element displayed:
```
lutris/gui/lutriswindow.py:696:        self.game_store.modelfilter.refilter()
lutris/gui/lutriswindow.py:711:        self.game_store.modelfilter.refilter()
```

**Expected behavior**
There shouldn't be any errors when selecting a runner that doesn't have any entries.

**Steps to reproduce**
1. Open Lutris from the command line.
2. Select a runner that has some visible entries
3. Select a runner that doesn't have any visible entries (the errors will be displayed)

**Comment by strycore** [comment]
The diagnostic is incorrect. I get error messages in the consoles even when switching to runner that have games.
It's also not possible to get the number of elements in the view before calling refilter since that's the method updating the items in the TreeModelSort

**Comment by strycore** [comment]
Closed by #3089

The TreeModelFilter has been completely removed. 
This is a bug in Gtk, ending in a completely cryptic comment: https://gitlab.gnome.org/GNOME/gtk/-/issues/1150
Since there is little chance we can solve that on the python side, I just delegated the sorting and filtering to the database.
